src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/ScannedDocumentsParser.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.hmcts.reform.bulkscanccdeventhandler.services;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.CcdCollectionElement;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.ResultOrErrors;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.ScannedDocument;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.services.exception.CallbackProcessingException;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.services.exception.CcdDataParseException;
+
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Collections.emptyList;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.ExceptionRecordFieldNames.SCANNED_DOCUMENTS;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.ResultOrErrors.result;
+
+/**
+ * Converts raw exception record data representing scanned documents
+ * (in the form of a list of maps of objects) into a strongly typed collection.
+ */
+@Service
+public class ScannedDocumentsParser {
+
+    private static final Logger log = LoggerFactory.getLogger(OcrDataParser.class);
+
+    private final CcdCollectionParser ccdCollectionParser;
+
+    public ScannedDocumentsParser(CcdCollectionParser ccdCollectionParser) {
+        this.ccdCollectionParser = ccdCollectionParser;
+    }
+
+    public ResultOrErrors<List<CcdCollectionElement<ScannedDocument>>> parseScannedDocuments(
+        Map<String, Object> exceptionRecordData
+    ) {
+        Object scannedDocumentsElement = exceptionRecordData.get(SCANNED_DOCUMENTS);
+
+        if (scannedDocumentsElement == null) {
+            return result(emptyList());
+        } else {
+            try {
+                List<CcdCollectionElement<ScannedDocument>> scannedDocuments =
+                    ccdCollectionParser.parseCcdCollection(scannedDocumentsElement, ScannedDocument.class);
+
+                return result(scannedDocuments);
+            } catch (CcdDataParseException e) {
+                throw new CallbackProcessingException(
```

Comment:
looks like it's either a result or exception is thrown. in that case, is `ResultOrErrors` 'wrapper' is needed?

Reply:
There are two levels of errors:
- The kind of 404/validation errors, which should result with a proper error list displayed to the user
- This kind, which represents an internal error. What's happening here is we're failing to process the request. Instead of a 200 response with errors and warnings we'll return a 500 response and CCD will display a generic error, which is the most adequate to the situation. In the future we'll need to move this bit to the library, so that transformer isn't even involved if the request is malformed.

Reply:
On a second thought, that's not great, either. After all, the request is invalid.

Reply:
Ok, in a following PR I'll map this exception (and possibly rename it) to the right response, so that CCD will return a generic error to the user.